eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceClientProvider.cs

```diff
@@ -444,11 +444,11 @@ protected override MethodProvider[] BuildMethods()
             // Only generate tag methods if the resource model has tag properties, has get and update methods
             if (HasTags() && _resourceMetadata.Methods.Any(m => m.Kind == ResourceOperationKind.Get) && updateMethodProvider is not null)
             {
-                // Find the update method to get its rest client
-                var updateMethod = _resourceMetadata.Methods.FirstOrDefault(m => m.Kind == ResourceOperationKind.Update || m.Kind == ResourceOperationKind.Create);
-                if (updateMethod is not null)
+                // Find the update input client, which should have both get and update methods
+                var client = GetUpdateClient();
+                if (client is not null)
                 {
-                    var updateRestClientInfo = _clientInfos[updateMethod.InputClient];
+                    var updateRestClientInfo = _clientInfos[client];
```

Comment:
so all these stuff is here to determine the corresponding rest client provider for this updateMethodProvider?
Instead of doing this stuff, how about we just use another local variable to store it, and here we could do:
```
if (HasTags() && _resourceMetadata.Methods.Any(m => m.Kind == ResourceOperationKind.Get) && updateMethodProvider is not null && updateRestClient is not null)
```

Reply:
left a TODO for struct